src/components/Dropdown/Dropdown.tsx

```diff
@@ -239,11 +239,14 @@ export const Dropdown: FC<DropdownProps> = React.memo(
           { [child.props.className]: child.props.className },
           { [styles.disabled]: disabled },
         ]);
+        if (child.props.id && dropdownReferenceId !== child.props.id) {
+          setReferenceElementId(child.props.id);
```

Comment:
This method to update the state is async, but in the code below, you are assuming that the state will have been updated. Is this still safe? I'm still getting used to the react lifecycle, but it seems like the below code _may_ have the update or may not (depending on how react queues the state update), but also that setting the state would also trigger another render, in which case it would pick up the correct value the second time?

Reply:
The string compare takes care of this and ensures the state is only ever set one time.

Reply:
`getReference` method is not async and the component is memoized.

Reply:
Added a comment clarifying the reason behind the compare.